### PR TITLE
Added "Copy as text" button to changelog page

### PR DESCRIPTION
**Pull request by MacDisein**
This is the pull request for issue [0021837](https://mantisbt.org/bugs/view.php?id=0021837): Text-based change log

### CODE REVIEW

**Comment by atrol** [comment]
> server based generation

+1 for that because
- loading two additional scripts on every page, that is just needed for a single operation on a single page is overhead
- maintaining the JS code if someone changes the backend code can be avoided

File: core/html_api.php [hunk 1]
```diff
@@ -327,12 +327,18 @@ function html_head_javascript() {
...
+		html_javascript_cdn_link( 'https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/' . CLIPBOARD_VERSION . '/clipboard.min.js', CLIPBOARD_HASH );
 	} else {
 		# JQuery
 		html_javascript_link( 'jquery-' . JQUERY_VERSION . '.min.js' );
 
 		# Dropzone
 		html_javascript_link( 'dropzone-' . DROPZONE_VERSION . '.min.js' );
+
+		# Clipboard
+		html_javascript_link( 'clipboard-' . CLIPBOARD_VERSION . '.min.js' );
```
Comment (dregad) [inline core/html_api.php]: Script does not load with cdn_enabled = OFF 
```
Loading failed for the <script> with source “http://localhost/mantis/js/clipboard-2.0.4.min.js”. changelog_page.php:25:1
```
Included file is `clipboard.min.js`

File: changelog_page.php [hunk 2]
```diff
@@ -149,6 +149,9 @@ function print_version_footer( $p_version_id, $p_issues_resolved ) {
 		 '&' . FILTER_PROPERTY_HIDE_STATUS . '=' . META_FILTER_NONE . '">';
 	echo lang_get( 'view_bugs_link' );
 	echo '</a>';
+	echo '<div class="btn-toolbar pull-right">';
+	echo '<a class="btn btn-xs btn-primary btn-white btn-round clp" data-tooltip-success="' . lang_get('copy_as_text_successful') . '" data-tooltip-failed="' . lang_get('copy_as_text_failed') . '">' . lang_get( 'copy_as_text' ) . '</a>';
```
Comment (dregad) [inline changelog_page.php]: Break long line

File: core/layout_api.php [hunk 3]
```diff
@@ -334,6 +334,7 @@ function layout_body_javascript() {
 
 	# ace theme scripts
 	html_javascript_link( 'ace.min.js' );
+	html_javascript_link( 'clipboard.impl.js' );
```
Comment (dregad) [inline core/layout_api.php]: This is not an ace theme script... Add a comment.

I would suggest to load this script only on changelog page. Currently it's loaded on every single page but it's not actually used anywhere else.

Finally, I would suggest renaming the script, e.g. `changelog.js`

File: lang/strings_german.txt [hunk 4]
```diff
@@ -1432,3 +1432,6 @@ $s_persist = 'Merken';
 $s_load = 'Laden';
 $s_apply_changes = 'Änderungen anwenden';
 $s_undo = 'Rückgängig';
+$s_copy_as_text = 'Als Text kopieren';
+$s_copy_as_text_successful = 'Kopiert';
+$s_copy_as_text_failed = 'Fehler beim Kopieren';
```
Comment (dregad) [inline lang/strings_german.txt]: Foreign language strings should be added via TranslateWiki.

**Comment by dregad** [comment]
@MacDisein are you planning to address the review comments ?

**Comment by dregad** [comment]
Closing the PR due to lack of response from @MacDisein

**Pull request closed without merging**